api/edge_api/identities/serializers.py

```diff
@@ -45,7 +48,9 @@ def save(self, **kwargs):
             raise ValidationError(
                 f"Identity with identifier: {identifier} already exists"
             )
-        EdgeIdentity.dynamo_wrapper.put_item(build_identity_dict(self.instance))
+        EdgeIdentity.dynamo_wrapper.put_item(
+            map_engine_identity_to_identity_document(self.instance)
```

Comment:
I'm keen to understand why we had to change this? My understanding was that we were keeping the `build_identity_dict` functions for backwards compatibility, but if we've broken them - are there any issues elsewhere? e.g. python client?

Reply:
> but if we've broken them
We have, sadly. However, I've checked all the Python code we have and `build_identity_dict` has only been used in Core API. Those two calls are the last ones. I suggest we deprecate `build_identity_dict` completely after we eliminate them.

Reply:
Hmm... do we know why it's broken? Is it an easy fix? Just conscious we already released the major version so I guess our options are: 

1. Fix it
2. Release another major version to remove it entirely
3. (Slightly skanky) Remove it from the current major version

Reply:
It's broken because Marshmallow dumps datetimes to ISO-formatted string, while Pydantic's `.dict()` method doesn't.

I'm not in favour of option 1 because the functionality is only needed for Core API. 

Option 3 is dodgy but we did promise to deprecate the `builders` modules in 4.0.0 release notes. Maybe we could get away with a minor release.

Reply:
I suggest we move this over to https://github.com/Flagsmith/flagsmith-engine/issues/175